### PR TITLE
Truncate ecosystem comment when necessary

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR ensures that the PR comment action always creates a comment, even if the ecosystem results are huge by:

* Testing on the outcome of the step rather the comment text to prevent an out of memory error
* Use another PR comment plugin that automatically truncates the text if it exceeds GitHubs limit of 65536 characters. 

It would be nice if we could add a custom message when the results are truncated but the plugin does not support that. I think this is already a huge improvement because it prevents that we loose the ecosystem result when it's most needed

## Test Plan

I commented out the traversal of the class body in checker to generate a large regression. 

* The [PR Comment run](https://github.com/astral-sh/ruff/actions/runs/5990006807) with the workflow from main fails 
* The manually scheduled [PR Comment run](https://github.com/astral-sh/ruff/actions/runs/5990009960/job/16246859872) on this branch succeeds

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #6899** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/6899" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/6899?utm_source=stack-comment).

File: .github/workflows/pr-comment.yaml [hunk 1]
```diff
@@ -47,19 +47,30 @@ jobs:
...
-        uses: thollander/actions-comment-pull-request@v2
+        if: steps.find-comment.outcome == 'success'
+        uses: peter-evans/create-or-update-comment@v3
         with:
-          pr_number: ${{ steps.pr-number.outputs.pr-number }}
-          message: ${{ steps.generate-comment.outputs.comment }}
-          comment_tag: PR Check Results
+          comment-d: ${{ steps.find-comment.outputs.comment-id }}
+          issue-number: ${{ steps.pr-number.outputs.pr-number }}
+          body-path: comment.txt
```
Comment (MichaReiser) [inline .github/workflows/pr-comment.yaml:75]: It needs to be a file to avoid OOMing when loading the content

**Review by charliermarsh — APPROVED** [review]
Thank you!

Is "comment << EOF" intended?

Screen Shot 2023-08-27 at 9 52 16 AM

**Comment by MichaReiser** [comment]
> Is "comment << EOF" intended?

Sure :laughing:

**Pull request merged**